test/hotspot/jtreg/runtime/ErrorHandling/TestSigInfoInHsErrFile.java

```diff
@@ -0,0 +1,139 @@
+import jdk.test.lib.process.ProcessTools;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.regex.Pattern;
+
+public class TestSigInfoInHsErrFile {
+
+  public static void main(String[] args) throws Exception {
+    if (args.length != 1) {
+      throw new IllegalArgumentException("missing argument");
+    } else if (args[0].equals("segv")) {
+      testWithSEGV();
+    } else if (args[0].equals("fpe")) {
+      testWithFPE();
+    } else {
+      throw new IllegalArgumentException("unknown argument");
+    }
+  }
+
+  static void testWithFPE() throws Exception {
+
+    ProcessBuilder pb = ProcessTools.createJavaProcessBuilder(
+            "-XX:+UnlockDiagnosticVMOptions",
+            "-Xmx100M",
+            "-XX:-CreateCoredumpOnCrash",
+            "-XX:ErrorHandlerTest=15",
+            "-version");
+
+    OutputAnalyzer output = new OutputAnalyzer(pb.start());
+    output.shouldNotHaveExitValue(0);
+
+    // we should have crashed with a SIGFPE
+    output.shouldMatch("#.+SIGFPE.*");
+
+    // extract hs-err file
+    File f = HsErrFileUtils.openHsErrFileFromOutput(output);
+
+    ArrayList<Pattern> patterns = new ArrayList<>();
+    patterns.add(Pattern.compile("# A fatal error has been detected.*"));
+    patterns.add(Pattern.compile("# *SIGFPE.*"));
+    patterns.add(Pattern.compile("# *Problematic frame.*"));
+    patterns.add(Pattern.compile("# .*VMError::controlled_crash.*"));
+
+    patterns.add(Pattern.compile("siginfo: si_signo: \\d+ \\(SIGFPE\\), si_code: \\d+ \\(FPE_INTDIV\\).*"));
+
+    HsErrFileUtils.checkHsErrFileContent(f, patterns.toArray(new Pattern[] {}), true);
+
+  }
+
+  static void testWithSEGV() throws Exception {
+
+    ProcessBuilder pb = ProcessTools.createJavaProcessBuilder(
+        "-XX:+UnlockDiagnosticVMOptions",
+        "-Xmx100M",
+        "-XX:-CreateCoredumpOnCrash",
+        "-XX:ErrorHandlerTest=14",
+        "-version");
+
+    OutputAnalyzer output = new OutputAnalyzer(pb.start());
+    output.shouldNotHaveExitValue(0);
+
+    // we should have crashed with a SIGFPE
+    output.shouldMatch("#.+SIGSEGV.*");
+
+    // extract hs-err file
+    File f = HsErrFileUtils.openHsErrFileFromOutput(output);
+
+    ArrayList<Pattern> patterns = new ArrayList<>();
+    patterns.add(Pattern.compile("# A fatal error has been detected.*"));
+    patterns.add(Pattern.compile("# *SIGSEGV.*"));
+    patterns.add(Pattern.compile("# *Problematic frame.*"));
+    patterns.add(Pattern.compile("# .*VMError::controlled_crash.*"));
+
+    if (!Platform.isAix()) {
+      // Crash address is, for all non-AIX platforms, at 1K (0x400). See VMError::_segfault_address
+      patterns.add(Pattern.compile("siginfo: si_signo: \\d+ \\(SIGSEGV\\), si_code: \\d+ \\(SEGV_MAPERR\\), si_addr: 0x0*400"));
+    } else {
+      patterns.add(Pattern.compile("siginfo: si_signo: \\d+ \\(SIGSEGV\\).*"));
```

Comment:
Why not use the AIX 5K address here?

Reply:
Oversight. I have no way to test AIX anymore since we handed porting over to IBM.